### DISCUSSION TITLE
Capturing Server TLS Certificates

### DISCUSSION THREAD

**Discussion by aaronst** [encode/httpx#2089] [Q&A] [answered]
Is it possible to get the server TLS certificate for any given `Response`? `aiohttp` makes this possible with access to the `asyncio.Transport`, which has the [`.get_extra_info()`](https://docs.python.org/3.10/library/asyncio-protocol.html#asyncio.BaseTransport.get_extra_info) method. It doesn't look like that's used in `httpx` though.

**Comment by lovelydinosaur** [comment, marked as answer]
Great question yup. We need some docs around this.

Take a look at `response.extensions['network_stream']`. This exposes a `get_extra_info()`, which includes SSL info in some contexts.

The low-level `httpcore` package has some docs about this https://www.encode.io/httpcore/extensions/#network_stream which are also valid to `httpx` since the extensions are passed through.

**Reply by aaronst** [reply]
Awesome yeah, I did end up finding this but forgot to come back here and answer. Thanks!
